R/expect-setequal.R

```diff
@@ -32,41 +32,37 @@ expect_setequal <- function(object, expected) {
   }
 
   act_miss <- !act$val %in% exp$val
-  if (any(act_miss)) {
-    fail(
-      paste0(act$lab, "[", locations(act_miss), "] absent from ", exp$lab)
-    )
-  }
-
   exp_miss <- !exp$val %in% act$val
-  if (any(exp_miss)) {
-    fail(
-      paste0(exp$lab, "[", locations(exp_miss), "] absent from ", act$lab)
-    )
-  }
 
-  if (!any(exp_miss) && !any(act_miss)) {
+  if (any(exp_miss) || any(act_miss)) {
+    fail(paste0(
+      act$lab, " and ", exp$lab, " don't have the same values\n",
+      if (any(exp_miss))
+        paste0("* Missing from expected: ",  strings(act$val[act_miss]), "\n"),
+      if (any(act_miss))
+        paste0("* Missing from actual: ",  strings(exp$val[exp_miss]), "\n")
+    ))
+  } else {
     succeed()
   }
 
   invisible(act$val)
 }
 
-is_vector <- function(x) is.list(x) || (is.atomic(x) && !is.null(x))
-
-locations <- function(i) {
-  loc <- which(i)
-  if (length(loc) == 1) {
-    return(loc)
+strings <- function(x) {
```

Comment:
Can't you use `expect_setequal()` on more than strings though? How does this look on numeric values?

Reply:
Oops, yes.